schema/signature.go

```diff
@@ -25,3 +27,20 @@ type FunctionSignature struct {
 	// parameter if it is supported.
 	VarParam *function.Parameter
 }
+
+func (fs *FunctionSignature) Copy() *FunctionSignature {
+	newFS := &FunctionSignature{
+		Description: fs.Description,
+		Detail:      fs.Detail,
+		ReturnType:  fs.ReturnType,
+	}
+
+	copy(newFS.Params, fs.Params)
```

Comment:
```suggestion
        newFS.Params = make([]function.Parameter, len(fs.Params))
	copy(newFS.Params, fs.Params)
```

We still need to create the slice before copying something into it.

Reply:
Well that makes sense 😅